File: 032_Staticke metode/src/hr/unipu/java/topic3/staticmethods/Main.java
package hr.unipu.java.topic3.staticmethods;

public class Main {

    public static void main(String[] args) {
        Point a = new Point(0,0);
        Point b = new Point(6,0);
        Point c = new Point(3,5);
        Point center = Point.centerOf(a, b, c); // poziv statičke metode
        System.out.println(center.toString());  //toString ovdje nije bio nužan, ali je dodan da se naglasi razlika između statičke i ne-statičke

        // Koristiti polje kao argument nije uvijek praktično, jer ga prije poziva treba formirati.
        Point[] points = new Point[] {a, b, c};
        center = Point.centerOf(points);
        System.out.println(center);

        // Primjer poziva metode s varijabilnim brojem argumenata
        System.out.println(Point.centerOf(a, b));
        System.out.println(Point.centerOf(a, b, c));
        System.out.println(Point.centerOf(a, b, c, new Point(4,8)));

        // 2D vektor pomoću klase Point
        Vector v = new Vector(new Point(3,4));  //klasa Vector za 2D-vektor koji predstavlja vektor iz ishodišta do točke predane u konstruktoru
        System.out.println(v);
        // Promjena iz kanonske baze u neku novu bazu
        Vector.alpha1 = new Vector(0,1);
        Vector.alpha2 = new Vector(-1,2);
        System.out.println(v);
    }
}

File: 032_Staticke metode/src/hr/unipu/kotlin/topic3/staticmethods/Main.kt
package hr.unipu.kotlin.topic3.staticmethods

fun main(args: Array<String>) {
    val a = Point(0.0, 0.0)
    val b = Point(6.0, 0.0)
    val c = Point(3.0, 5.0)
    var center = Point.centerOf(a, b, c) // poziv statičke metode
    println(center.toString())  //toString ovdje nije bio nužan, ali je dodan da se naglasi razlika između statičke i ne-statičke
    // > (3.0, 1.6666666666666667)


    // Koristiti polje kao argument nije uvijek praktično, jer ga prije poziva treba formirati.
    val points = arrayOf(a, b, c)
    center = Point.centerOf(points)
    println(center)
    // > (3.0, 1.6666666666666667)

    // Primjer poziva metode s varijabilnim brojem argumenata
    println(Point.centerOf(a, b))
    println(Point.centerOf(a, b, c))
    println(Point.centerOf(a, b, c, Point(4.0, 8.0)))
    // > (3.0, 0.0)
    // > (3.0, 1.6666666666666667)
    // > (3.25, 3.25)

    // 2D vektor pomoću klase Point
    val v = Vector(Point(3.0, 4.0))  //klasa Vector za 2D-vektor koji predstavlja vektor iz ishodišta do točke predane u konstruktoru
    println(v)
    // > (3,00, 4,00) = 3,00 * (1,00, 0,00) + 4,00 * (0,00, 1,00)

    // Promjena iz kanonske baze u neku novu bazu
    Vector.alpha1 = Vector(0.0, 1.0)
    Vector.alpha2 = Vector(-1.0, 2.0)
    println(v)
    // > (3,00, 4,00) = 10,00 * (0,00, 1,00) + -3,00 * (-1,00, 2,00)

}
